## 1. Introdução

O presente exercício consiste na análise de uma linha de transmissão trifásica em duas configurações distintas: linha sem transposição e linha com transposição.

## 2. Dados da rede

A Figura 1 apresenta a matriz de impedâncias série da linha sem transposição. Note que somente o triângulo superior da matriz é indicado, já que a matriz é simétrica.



![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/matrizdeimpedanciasseriedalinhasemtransposicaoemohmporkm.png?raw=true.png)

### Dados da rede
* Comprimento da linha = 200 km
* Corrente absorvida pela carga equilibrada, alimentada pela linha. Em [A], em cada fase = 200A

### 3. Questões

* 1) Calcular o módulo da queda de tensão por fase na linha sem transposição, em [V] (3 valores).

* 2) Calcular a perda ativa total (3 fases) na linha sem transposição, em [MW].

* 3) Calcular o módulo da queda de tensão por fase na linha completamente transposta, em [V] (3 valores).

* 4) Calcular a perda ativa total (3 fases) na linha completamente transposta, em [MW].

In [5]:
import cmath
import numpy as np
K = 1000
M = K*K


In [6]:
Z = 200*np.array([[0.0846 + 0.6423j , 0.0567 + 0.3308j , 0.0573 + 0.3499j],
    [0.0567 + 0.3308j , 0.0833 + 0.6438j , 0.0567 + 0.3308j],
    [0.0573 + 0.3499j ,  0.0567 + 0.3308j , 0.0846 + 0.6423j]])


In [7]:
def phase(x): # em graus
    return ((180/np.pi)*(np.arctan2(x.imag,x.real)))

def modulo(x):
    return np.sqrt(x.real*x.real + x.imag*x.imag)
            
def polar(x):
    print(str(modulo(x)) + "/_" + str(phase(x)))
    
def cart2pol(x, y):
    rho = np.sqrt(x**2 + y**2)
    phi = 180*np.arctan2(y, x)/np.pi
    return(rho, phi)

def pol2cart(rho, phi):
    x = rho * np.cos(phi)
    y = rho * np.sin(phi)
    return x + y*1j 


In [8]:
alpha = -0.4999999999999998 + 0.8660254037844387j

In [9]:
I = 200*np.array([1 , alpha**2, alpha])
polar(I)

[200. 200. 200.]/_[   0. -120.  120.]


In [10]:
polar(alpha)

0.9999999999999999/_119.99999999999999


# 1) Calculo do modulo da queda de tensão na linha por fase (sem transposição de fases) 

In [11]:
V = Z@I

In [12]:
V

array([   442.35659151+12098.78460969j,  10310.63805538 -7181.45102963j,
       -11324.67653115 -4499.51564938j])

In [13]:
V.shape

(3,)

In [14]:
polar(V)

[12106.86864494 12565.13016248 12185.80894378]/_[  87.90608049  -34.85755422 -158.33110683]


#### Questão 1
* VAA' = 12106.86864494 /_ 87.90608049
* VBB' = 12565.13016248 /_ -34.85755422
* VCC' = 12185.80894378 /_ -158.33110683

In [15]:
modulo(V)

array([12106.86864494, 12565.13016248, 12185.80894378])

In [16]:
def diferenca_de_phase(a,b):
    return abs(b - a)
def dif(a,b):
    print(diferenca_de_phase(phase(V[a]),phase(V[b])))

In [17]:
print(-34.85755422+34.85755422)
print(-158.33110683+34.85755422)
print(87.90608049+34.85755422)

0.0
-123.47355261000001
122.76363470999999


Vemos que as diferenças de fase que deveriam ser exatamente 120 graus
não bate com a realidade(-123 e 122). Sendo assim propõe-se a  transposição de fase
a ser estudada nos itens 3) e 4)



In [18]:
#Erro:
EBBlinha = 100*(123.47355261000001-120)/120 
ECClinha = 100*(122.76363470999999-120)/120
print("Erro na linha B = "+ str(EBBlinha) + " %")
print("Erro na linha C = "+ str(ECClinha) + "%")

Erro na linha B = 2.8946271750000108 %
Erro na linha C = 2.3030289249999925%


# 2) Calculo da potência (sem transposição de fases)

In [19]:
def FP(phaseV,phaseI):
    return np.cos(np.pi*(phaseV - phaseI)/180)
P = (V*I.conjugate()).real/M
print(P) #potencia ativa em MW

[0.08847132 0.2128     0.35312868]


In [20]:
print("Perda Ativa total = " + str(P[0] + P[1] + P[2]) +" " + "MW")

Perda Ativa total = 0.6543999999999993 MW


Assim vemos que
* PAA' = 0.08847132 MW
* PBB' = 0.2128 MW
* PCC' = 0.35312868

quando na verdade deveriamos ter P0 = P1 = P2

# 3) Calculando o modulo da queda de tensão na linha por fase (com transposição de fase)


 ![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/TransposicaoDefase1.png?raw=true.png)

Vemos que nesse caso cada mutua na matriz Z afetaria o sistema multiplicada por um fator 1/3. Alem disso todas fases
AA', BB', CC' experimentariam as mesmas Mutuas, o que tenderia a equilibrar mais o sistema com relacção a parte imaginaria
dos termos que não se encontram na diagonal principal da matriz Z. Porém vejamos que efeito isso tem na parte real desses
valores haja vista que eles modelam não as mutuas mas sim efeitos de perda por retorno pela terra que é composta por
diferentes densidades e extratos nos 200 km de linha considerados.

In [21]:
#Mutua média ZM , A[LINHA][COLUNA]
# ZM = (MAB + MAC + MBC)/3
ZM = 200*((0.0567 + 0.3308j) + (0.0573 + 0.3499j) + (0.0567 + 0.3308j))/3

In [22]:
ZM

(11.38+67.43333333333332j)

In [23]:
ZA = (0.0846 + 0.6423j)*200
ZB = (0.0833 + 0.6438j)*200
ZC = (0.0846 + 0.6423j)*200
ZP = (ZA + ZB + ZC)/3
Z_com_transposicao = np.array([[ZP,ZM,ZM],[ZM,ZP,ZM],[ZM,ZM,ZP]])

In [24]:
ZP/200

(0.08416666666666667+0.6428000000000001j)

In [25]:
Z_com_transposicao

array([[16.83333333+128.56j      , 11.38       +67.43333333j,
        11.38       +67.43333333j],
       [11.38       +67.43333333j, 16.83333333+128.56j      ,
        11.38       +67.43333333j],
       [11.38       +67.43333333j, 11.38       +67.43333333j,
        16.83333333+128.56j      ]])

In [26]:
V_com_transposicao = Z_com_transposicao@I
polar(V_com_transposicao)

[12273.88809175 12273.88809175 12273.88809175]/_[  84.90193083  -35.09806917 -155.09806917]


In [27]:
modulo(V_com_transposicao)

array([12273.88809175, 12273.88809175, 12273.88809175])

In [28]:
print(-35.09806917 +35.09806917 )
print(-155.09806917+35.09806917)
print(84.90193083+35.09806917)

0.0
-120.0
120.0


In [29]:
#Erro:
EBBlinha = 100*(120.0-120)/120 
ECClinha = 100*(120.0-120)/120
print("Erro na linha B = "+ str(EBBlinha) + " %")
print("Erro na linha C = "+ str(ECClinha) + "%")

Erro na linha B = 0.0 %
Erro na linha C = 0.0%


# 4) Potência Ativa total com transposição

In [30]:
P = (V_com_transposicao*I.conjugate()).real/M
print(P) #potencia ativa em MW

[0.21813333 0.21813333 0.21813333]


In [31]:
print("Perda Ativa total = " + str(P[0] + P[1] + P[2]) +" " + "MW")

Perda Ativa total = 0.6543999999999994 MW


##  Questões desafio
* Questão 1
    Por que na matriz de impedâncias série da linha sem transposição as impedâncias mútuas possuem parte
    real não nula?
    
    
* Questão 2
    Por que na matriz de impedâncias série da linha sem transposição as impedâncias próprias são diferentes,
    se a linha foi construída com 3 condutores idênticos?
    
    
* Questão 3
    O que se pode afirmar em relação à disposição geométrica dos condutores, a partir dos valores numéricos
    na matriz de impedâncias série da linha sem transposição?
    
    
* Questão 4
    Uma linha de alta tensão possui, além dos condutores de fase, um ou mais condutores guarda, cuja
    finalidade é proteger a linha contra sobretensões oriundas de descargas atmosféricas (diretamente na linha
    ou em suas proximidades).
    Considere que no presente caso a linha conta com 2 cabos guarda, de tal forma que a matriz de
    impedâncias série completa (5 condutores) é dada por (valores em ohm/km):(Figura 1)
    
    
    Os cabos guarda podem ser aterrados somente numa extremidade, ou então aterrados nas duasextremidades. A condição de contorno em cada caso é dada por:
* a) cabos guarda aterrados numa extremidade:(Figura 2)
* b) cabos guarda aterrados nas duas extremidades:(Figura 3)


    Pede-se determinar a matriz de impedâncias série equivalente (3 x 3) após a eliminação dos cabos guarda,
    nos 2 casos (cabos guarda aterrados numa extremidade e cabos guarda aterrados nas duas extremidades).
    Sugestão:
    Lembre que a matriz de impedâncias série representa na verdade um sistema de equações; no presente
    caso de 5 condutores tem-se:(Figura 4)
    
    
    Particione a matriz segundo a terceira linha/coluna e imponha cada uma das duas condições de contorno;
    escreva as duas equações com as submatrizes que resultam da partição, e obtenha finalmente uma matriz
    3 x 3 modificada (a matriz equivalente que está sendo solicitada).




![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/matrizesdodesafiotransposicao.png?raw=true.png)

# 1) Desafio

As impedâncias fora da diagonal principal modelam efeitos de indutâncias mutuas dadas por reatâncias representadas pela parte imaginária dessas impedâncias. Sabe-se que Indutâncias mutuas não possuem parte real em sua representação por impedâncias. Sendo assim essa parte realse deve ao retorno via solo.


# 2) Desafio
Embora os três fios sejam identicos é preciso considerar que existem ligeiras diferenças em seu caminho que podem ter ocasionado isso:
##### Parte Resisitiva 
1) Resistência na emenda dos fios 
##### Parte Reativa 
1) Diferenças no solo abaixo da passagem da rede elétrica entre as fases de tal modo que a "espira" fechada entre uma dada fase e o solo em comparação com outro par fase-solo seja diferente.
* Extrato do solo
* Materias diferentes abaixo de cada fase
* **Diferenças de altura entre um cabo e o solo devido a geometria da torre de transmissão**

Com tudo observa-se que essas diferenças, apesar de indesejadas , ainda ficam na ordem de deca-ohms.

In [34]:
RA = Z[0][0].real
RB = Z[1][1].real
RC = Z[2][2].real

XA = Z[0][0].imag
XB = Z[1][1].imag
XC = Z[2][2].imag
print("Resistenências = " + str(RA) +" ; "+ str(RB) +" ; "+ str(RC))
print("Reatâncias = " + str(XA) +" ; "+ str(XB) +" ; "+ str(XC))

Resistenências = 16.919999999999998 ; 16.66 ; 16.919999999999998
Reatâncias = 128.46 ; 128.76000000000002 ; 128.46


# 3) Desafio
A reatância de cada fase nos dará uma idea boa da distribuição geométrica dos fios dado que a Indutância prórpia nesse caso dependerá de fatores geométricos da espira formada entre o cabo e o solo. 

Matemáticamente temos que:

L = N²/***R*** [H]

***R*** = lm/***u***.S [H-¹]

S = "Área da espira" = h.L_cabo [m]

h = altura do fio ; L_cabo = comprimento do cabo

Ou seja, considerando-se sigma = N².***u***.L_cabo / lm

Temos uma formula para L em função da altura h:

                            L = sigma.h

In [37]:
#Sendo assim tempos que:
#(Assumindo-se a rede brasileira em 60 Hz)
f = 60
w  = 2*np.pi*f
LA = XA/w
LB = XB/w
LC = XC/w
print("Indutâncias prórpias " + str(LA) +" ; "+ str(LB) +" ; "+ str(LC))
print("Indutâncias prórpias normalizadas " + str(100*LA/LB) +" ; "+ str(100*LB/LB) +" ; "+ str(100*LC/LB))

Indutâncias prórpias 0.34075073315974796 ; 0.34154650787520746 ; 0.34075073315974796
Indutâncias prórpias normalizadas 99.76700838769804 ; 100.0 ; 99.76700838769804


Assim conclui-se que as fases A e C são lançadas a uma mesma altura ao passo que a fase B é lançada a uma altura superior as outras duas descrita pela relação linear entre indutância prórpia e altura do cabo descrita acima.
(Para essa analise assumiu-se que todos os parâmetros que compõe *sigma* fossem aproximadamente constantes.

![](https://github.com/Zolubas/Eletric-Power-Systems/blob/master/tiposdetorresdetransmissao.png?raw=true.png)

De uma forma visual, poderiamos concluir que dentre esses tipos de torre para linha de transmissão a torre utilizada seria aquela destacada a esquerda.
https://www.abcem.org.br/construmetal/2016/downloads/apresentacao/18_TORRE-DE-TRANSMISSAO-DE-ENERGIA-ELETRICA.pdf

# 4) Desafio
